core/jvm/src/test/scala/fs2/MemorySanityChecks.scala

```diff
@@ -227,3 +227,18 @@ object ZipThenBindThenParJoin extends App {
     .drain
     .unsafeRunSync()
 }
+
+object RetrySanityTest extends App {
```

Comment:
Unrelated to this PR but I think this repetition (cs, timer, etc) could be extracted into a common `SanityApp` or smth, no?

Reply:
Yeah, these started as minimizations and hence each were as small as possible. IOApp has too much overhead due to the cancellation stuff. I’d like to see these all refactored in to an integration test suite that we can run periodically. Maybe run each test for a minute or something, checking heap growth.